maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/NodeHelper.java

```diff
@@ -102,5 +125,20 @@ private void runFrontendMavenPlugin(String goal, List<MojoExecutor.Element> conf
 				executionEnvironment(mavenProject, mavenSession, pluginManager)
 		);
 	}
+
+	private File getExistingCache() {
+		String tempDirPath = System.getProperty("java.io.tmpdir");
+		File tempDir = new File(tempDirPath);
+		if (tempDir.exists() && tempDir.isDirectory()) {
+			File[] cache = tempDir.listFiles(file -> file.isDirectory() && file.getName().startsWith("openmrs-sdk-node"));
+			if (cache == null) {
+				return null;
+			}
+			if (cache.length > 0) {
```

Comment:
What happens to the rest of the caches?

Reply:
Should we delete them??

Reply:
If they are more than one, will the first one be consistently the same for subsequent calls?

Reply:
yes

Reply:
What do you suggest we should do here? We could delete the other caches.

Reply:
> yes

This actually isn't guaranteed. We're operating on files in the order returned from `listFiles()` which has this caveat:

"There is no guarantee that the name strings in the resulting array will appear in any specific order; they are not, in particular, guaranteed to appear in alphabetical order. "

Similarly, `createTempDirectory()` says:

"The details as to how the name of the directory is constructed is implementation dependent and therefore not specified."

In order to guarantee that this always returns the same folder, we'd need a guarantee that `listFiles()` returned files in a consistent order and that `createTempDirectory()` generates files in a defined order that will fit the order returned by `listFiles()`, but those details are specifically disclaimed.

(On Windows, the default JVM does indeed use the tick count as the source of randomness for `createTempDirectory()` and `listFiles()` likely returns in a consistent order, which would likely result in this appearing to work; however, Linux, the underlying `mkstemp()` implementation will try to use a RNG, falling back to ticks if not available on that device. I haven't looked into the details of macOS.)

Probably the thing to do is to make the `tempDir` a class variable instead of an instance variable so it's only set once per JVM, which would guarantee consistent results across multiple invocations of the NodeHelper within a single invocation of the SDK.